packages/analytics-core/src/config.ts

```diff
@@ -30,6 +33,7 @@ export const getDefaultConfig = () => ({
   serverUrl: AMPLITUDE_SERVER_URL,
   serverZone: 'US' as ServerZoneType,
   useBatch: false,
+  diagnosticProvider: new Diagnostic(),
```

Comment:
Since this is being added in core, it would be good to have some functional implementation for the `BaseDiagnostic`. Would it make sense to log these to the logger provider in the default (non-browser) case? Or will that duplicate messages?

Reply:
After discussion, we decided not to have functional implementation for `BaseDiagnostic` as it is shared by Node and RN but is disabled and haven't planned to implement them yet. Log to logger provider will double logging.